tests/basic/pure_test.go

```diff
@@ -2166,3 +2166,197 @@ func faLUNExists(faVolList []string, pvc string) bool {
+	})
+
+	stepLog = "Schedule apps, migrate apps from node 1 to node 2 and check if new multipath has been updated and old multipath has been erased"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+
+		//get device path of the volume
+		devicePaths := make([]string, 0)
+
+		stepLog = "Schedule fada deployment apps"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			// select a node for apps to be scheduled
+			applist := Inst().AppList
+			storageNodes := node.GetStorageNodes()
+			selectedNode := storageNodes[0]
+			secondNode := storageNodes[1]
+			log.Infof("Length of storage nodes: %v", len(storageNodes))
+			log.InfoD("Selected Node: %v", selectedNode.Name)
+			defer func() {
+				Inst().AppList = applist
+				err = Inst().S.RemoveLabelOnNode(selectedNode, "apptype")
+				log.FailOnError(err, "error removing label on node [%s]", selectedNode.Name)
+			}()
+			Inst().AppList = []string{"nginx-fada-deploy"}
+			err = Inst().S.AddLabelOnNode(selectedNode, "apptype", k8s.PureDAVolumeLabelValueFA)
+
+			log.FailOnError(err, fmt.Sprintf("Failed add label on node %s", selectedNode.Name))
+			Provisioner := fmt.Sprintf("%v", portworx.PortworxCsi)
+
+			stepLog = fmt.Sprintf("schedule application")
+			Step(stepLog, func() {
+				for i := 0; i < Inst().GlobalScaleFactor; i++ {
+					taskName := fmt.Sprintf("vol-migrate-test-%v", i)
+					context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+						AppKeys:            Inst().AppList,
+						StorageProvisioner: Provisioner,
+					})
+					log.FailOnError(err, "Failed to schedule application of %v namespace", taskName)
+					contexts = append(contexts, context...)
+				}
+				ValidateApplications(contexts)
+			})
+
+			stepLog = fmt.Sprintf("Check where the apps is scheduled and Stop Px on that node")
+			Step(stepLog, func() {
+				//get the volume name and inspect volume to get device path
+				for _, ctx := range contexts {
+					volumes, err := Inst().S.GetVolumes(ctx)
+					log.FailOnError(err, "Failed to get volumes for app %v", ctx.App.Key)
+					for _, volume := range volumes {
+						volInspect, err := Inst().V.InspectVolume(volume.ID)
+						log.FailOnError(err, "Failed to inspect volume %v", volume.ID)
+						devicePath := volInspect.DevicePath
+						// get part of the device path
+						log.Infof("device path: %v", devicePath)
+						devicePathSplit := strings.Split(devicePath, "/")
+						devicePath = devicePathSplit[len(devicePathSplit)-1]
+						devicePaths = append(devicePaths, devicePath)
+						log.InfoD("Device path of the volume: %v , device path: %v", volumes[0].Name, devicePath)
+					}
+				}
+
+				StopVolDriverAndWait([]node.Node{selectedNode})
+			})
+			defer func() {
+				err = core.Instance().UnCordonNode(selectedNode.Name, defaultCommandTimeout, defaultCommandRetry)
+				log.FailOnError(err, "Failed to uncordon node %v", selectedNode.Name)
+				log.Infof("uncordoned node %v", selectedNode.Name)
+
+				err = Inst().S.RemoveLabelOnNode(secondNode, "apptype")
```

Comment:
RemoveLabelOnNode is deferred twice I think . can u plz check

Reply:
This removelabel is for the second node where the apps are moved to. so this is required.